src/main/java/org/gridsuite/config/ConfigService.java

```diff
@@ -0,0 +1,136 @@
+        return () -> configUpdatePublisher.log(CATEGORY_BROKER_OUTPUT, Level.FINE);
+    }
+
+    @Autowired
+    public ConfigService(ConfigInfosRepository configRepository) {
+        this.configRepository = configRepository;
+    }
+
+    Mono<ConfigInfos> getConfigUiParameters(String userId) {
+        Mono<ConfigInfosEntity> configUiInfosEntityMono = configRepository.findByUserId(userId);
+        return configUiInfosEntityMono.map(ConfigService::toConfigUiInfos)
+                .switchIfEmpty(initialiseDefaultParameters(userId).map(ConfigService::toConfigUiInfos));
+    }
+
+    private Mono<ConfigInfosEntity> initialiseDefaultParameters(String userId) {
+        ConfigInfosEntity newConfigInfosEntity = ConfigInfosEntity.builder()
+                .userId(userId)
+                .substationLayout("horizontal")
+                .lineFlowMode("feeders")
+                .lineFlowColorMode("nominalVoltage")
+                .centerLabel(false)
+                .diagonalLabel(false)
+                .lineFlowAlertThreshold(100)
+                .lineFullPath(true)
+                .lineParallelPath(true)
+                .theme("Dark")
+                .useName(true)
+                .viewOverloadsTable(false)
+                .build();
+        return configRepository.save(newConfigInfosEntity);
+    }
+
+    Mono<ConfigInfos> updateParameters(String userId, ConfigInfos configInfos) {
+        Mono<ConfigInfosEntity> configInfosEntityMono = configRepository.findByUserId(userId);
+        return configInfosEntityMono.flatMap(configInfosEntity -> {
+            updateConfigUiInfosEntity(configInfosEntity, configInfos);
+            configUpdatePublisher.onNext(MessageBuilder.withPayload("")
+                    .setHeader(HEADER_USER_ID, userId)
+                    .build());
+            return configRepository.save(configInfosEntity);
+        }).map(ConfigService::toConfigUiInfos);
+    }
+
+    private static ConfigInfos toConfigUiInfos(ConfigInfosEntity configInfosEntity) {
+        return ConfigInfos.builder()
+                .substationLayout(configInfosEntity.getSubstationLayout())
+                .lineFlowMode(configInfosEntity.getLineFlowMode())
+                .lineFlowColorMode(configInfosEntity.getLineFlowColorMode())
+                .centerLabel(configInfosEntity.getCenterLabel())
+                .diagonalLabel(configInfosEntity.getDiagonalLabel())
+                .lineFlowAlertThreshold(configInfosEntity.getLineFlowAlertThreshold())
+                .lineFullPath(configInfosEntity.getLineFullPath())
+                .lineParallelPath(configInfosEntity.getLineParallelPath())
+                .theme(configInfosEntity.getTheme())
+                .useName(configInfosEntity.getUseName())
+                .viewOverloadsTable(configInfosEntity.getViewOverloadsTable())
+                .build();
+    }
+
+    private static void updateConfigUiInfosEntity(ConfigInfosEntity configInfosEntity, ConfigInfos configInfos) {
```

Comment:
why we don't use the builder ?

Reply:
because i only want to update the updated field and i don't want to erase a real value with a null

Reply:
ok